File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>mpp;
        for(int i=0;i<nums.size();i++){
            int n =  nums[i];
            int more = target-n;
            if(mpp.find(more)!=mpp.end()){
                return {mpp[more],i};
            }
            mpp[n]=i;
        }
        return {-1,-1};

    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for index,num in enumerate(nums):
            tar = target-num
            if tar in seen:
                return [seen[tar],index]
            seen[num] = index
        return [-1,-1]